anndata/_core/sparse_dataset.py

```diff
@@ -282,11 +283,21 @@ class BaseCompressedSparseDataset(ABC):
 
     def __init__(self, group: h5py.Group | ZarrGroup):
         type(self)._check_group_format(group)
-        self.group = group
+        self._group = group
 
     shape: tuple[int, int]
     """Shape of the matrix."""
 
+    @property
+    def group(self):
+        return self._group
+
+    @group.setter
+    def group(self, val):
+        raise TypeError(
+            f"Do not reset group on a {type(self)}.  Instead use `sparse_dataset` to make a new class."
+        )
+
```

Comment:
If we are going to cache this, we probably don't want people swapping out the group but this is technically a breaking change since this is something people can technically do at the moment.  On the other hand, this class is exported in experimental.

Reply:
I wouldn’t worry about breaking changes in experimental stuff